rule/exported.go

```diff
@@ -155,69 +151,51 @@ func (w *lintExported) lintFuncDoc(fn *ast.FuncDecl) {
 
 	kind := "function"
 	name := fn.Name.Name
-	isMethod := fn.Recv != nil && len(fn.Recv.List) > 0
-	if isMethod {
-		kind = "method"
-		recv := typeparams.ReceiverType(fn)
-
-		if !ast.IsExported(recv) && w.disabledChecks.PrivateReceivers {
-			return
-		}
-
-		if commonMethods[name] {
+	if isMethod := fn.Recv != nil && len(fn.Recv.List) > 0; isMethod {
+		if !w.mustCheckMethod(fn) {
 			return
 		}
 
-		switch name {
-		case "Len", "Less", "Swap":
-			sortables := w.file.Pkg.Sortable()
-			if sortables[recv] {
-				return
-			}
-		}
+		kind = "method"
+		recv := typeparams.ReceiverType(fn)
 		name = recv + "." + name
 	}
 
 	if w.disabledChecks.isDisabled(kind) {
 		return
 	}
```

Comment:
Shouldn't this if be moved before the previous `if` ?

Reply:
I asked very same question to myself when refactoring... I didn't find a nice way of moving the `if` because it depends on `kind`  that is potentially set in the body of previous one.